File: service/src/main/java/com/coopra/service/interfaces/ActivitiesInterface.java
package com.coopra.service.interfaces;

import com.coopra.data.DashboardActivityEnvelope;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import retrofit2.http.Url;

public interface ActivitiesInterface {
    @GET("me/activities/tracks.json?limit=50")
    Call<DashboardActivityEnvelope> getFeedTracks(@Query("oauth_token") String token);

    @GET
    Call<DashboardActivityEnvelope> getNextTracks(@Url String url, @Query("oauth_token") String token);
}

File: service/src/main/java/com/coopra/service/interfaces/ActivitiesInterface.kt
package com.coopra.service.interfaces

import com.coopra.data.DashboardActivityEnvelope
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Url

interface ActivitiesInterface {
    @GET("me/activities?limit=50")
    suspend fun getFeedTracks(@Query("oauth_token") token: String): DashboardActivityEnvelope

    @GET
    suspend fun getNextTracks(@Url url: String, @Query("oauth_token") token: String): DashboardActivityEnvelope
}